File: model/src/main/java/net/contrapt/auction/model/BidderSummary.java
package net.contrapt.auction.model;

import org.apache.commons.lang.builder.ReflectionToStringBuilder;
import org.apache.commons.lang.builder.ToStringStyle;

import java.math.BigDecimal;

/**
 * Created by msimmons on 10/10/14.
 */
public class BidderSummary {

    private Long id;

    private String name;

    private BigDecimal totalBids = BigDecimal.ZERO;

    private BigDecimal totalPayments = BigDecimal.ZERO;

    public BidderSummary(Long id, String name, BigDecimal totalBids, BigDecimal totalPayments) {
        this.id = id;
        this.name = name;
        this.totalBids = totalBids;
        this.totalPayments = totalPayments;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public BigDecimal getTotalBids() {
        return totalBids;
    }

    public BigDecimal getTotalPayments() {
        return totalPayments;
    }

    @Override
    public String toString() {
        return ReflectionToStringBuilder.toString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }

}

File: model/src/main/kotlin/net/contrapt/auction/model/BidderSummary.kt
package net.contrapt.auction.model

import java.math.BigDecimal

/**
 * Created by msimmons on 10/10/14.
 */
class BidderSummary(val id: Long, val name: String, totalBids: BigDecimal?, totalPayments: BigDecimal?) : AbstractModel() {

    val totalBids : BigDecimal
    val totalPayments : BigDecimal

    init {
        this.totalBids = totalBids ?: BigDecimal.ZERO
        this.totalPayments = totalPayments ?: BigDecimal.ZERO
    }
}
